authlib/jose/drafts/_jwe_algorithms.py

```diff
@@ -88,19 +97,27 @@ def deliver_at_recipient(self, recipient_key, sender_static_pubkey, sender_ephem
     def _generate_ephemeral_key(self, key):
         return key.generate_key(key['crv'], is_private=True)
 
-    def _prepare_headers(self, sender_key, epk):
+    def _prepare_headers(self, epk):
         # REQUIRED_JSON_FIELDS contains only public fields
         pub_epk = {k: epk[k] for k in epk.REQUIRED_JSON_FIELDS}
         pub_epk['kty'] = epk.kty
         return {'epk': pub_epk}
 
-    def generate_keys_and_prepare_headers(self, enc_alg, key, sender_key):
+    def generate_keys_and_prepare_headers(self, enc_alg, key, sender_key, preset=None):
         if not isinstance(enc_alg, CBCHS2EncAlgorithm):
             raise InvalidEncryptionAlgorithmForECDH1PUWithKeyWrappingError()
 
-        epk = self._generate_ephemeral_key(key)
-        cek = enc_alg.generate_cek()
-        h = self._prepare_headers(sender_key, epk)
+        if preset and 'epk' in preset:
+            epk = preset['epk']
+            h = {}
```

Comment:
Should we get the header from `preset`?

Reply:
No, because here we construct specific recipient's unprotected header.